src/webots/vrml/WbNode.cpp

```diff
@@ -2046,6 +2146,17 @@ QStringList WbNode::documentationBookAndPage(bool isRobot) const {
   return mModel->documentationBookAndPage();
 }
 
+const WbNode *WbNode::findRobotRootNode() const {
+  const WbNode *tmpNode = this;
+  while (tmpNode != NULL && tmpNode->nodeModelName() != "Robot")
```

Comment:
To avoid comparing directly with the `model name` we could introduce a `isRobot` function in WbNode that returns false and override it in WbRobot.